app/controllers/api/v2/courses_controller.rb

```diff
@@ -102,6 +102,12 @@ def update
           @course.save
           @course.course_subjects.each(&:save)
 
+          NotificationService::CourseSubjectsUpdated.call(
+            course: @course,
+            previous_subject_names: @previous_subject_names,
```

Comment:
Yep. Instance vars. WTF you may ask?
In the notification we need to show the user what the course subjects have changed from and to. Same deal with the course title. When a course's subjects change the course title changes dynamically to reflect the new subjects. We cannot use Rails Dirty API to (easily) to check if course associations have changed and how.